components/places/android/library/src/main/java/org/mozilla/places/PlacesConnection.kt

```diff
@@ -96,31 +126,38 @@ open class PlacesConnection(path: String, encryption_key: String? = null) : Auto
             LibPlacesFFI.INSTANCE.places_destroy_string(cstring)
         }
     }
-
-    companion object {
-        // Constants for use on VisitObservation visitType
-        /** This transition type means the user followed a link. */
-        const val VISIT_TYPE_LINK: Int = 1
-        /** This transition type means that the user typed the page's URL in the
-         *  URL bar or selected it from UI (URL bar autocomplete results, etc).
-         */
-        const val VISIT_TYPE_TYPED: Int = 2
-        // TODO: rest of docs
-        const val VISIT_TYPE_BOOKMARK = 3
-        const val VISIT_TYPE_EMBED = 4
-        const val VISIT_TYPE_REDIRECT_PERMANENT = 5
-        const val VISIT_TYPE_REDIRECT_TEMPORARY = 6
-        const val VISIT_TYPE_DOWNLOAD = 7
-        const val VISIT_TYPE_FRAMED_LINK = 8
-        const val VISIT_TYPE_RELOAD = 9
-    }
 }
 
 open class PlacesException(msg: String): Exception(msg)
 open class InternalPanic(msg: String): PlacesException(msg)
 open class UrlParseFailed(msg: String): PlacesException(msg)
 open class InvalidPlaceInfo(msg: String): PlacesException(msg)
 
+@SuppressWarnings("MagicNumber")
+enum class VisitType(val type: Int) {
+    /** This transition type means the user followed a link. */
+    LINK(1),
+    /** This transition type means that the user typed the page's URL in the
+     *  URL bar or selected it from UI (URL bar autocomplete results, etc).
+     */
+    TYPED(2),
+    // TODO: rest of docs
+    BOOKMARK(3),
+    EMBED(4),
+    REDIRECT_PERMANENT(5),
+    REDIRECT_TEMPORARY(6),
+    DOWNLOAD(7),
+    FRAMED_LINK(8),
+    RELOAD(9)
+}
+
+/**
+ * Encapsulates either information about a visit to a page, or meta information about the page, or both.
+ *
+ * @param visitType one of [VisitType] or [Nothing].
+ *  When [Nothing], this observation will not be considered a new visit and will instead update meta
+ *  information about the [url].
+ */
 data class VisitObservation(
     val url: String,
     val visitType: Int? = null,
```

Comment:
Can you make this take a `VisitType?` instead, and convert to the number in the toJSON?

(Also, I know I did this, but is null the right default here? Probably not – TBH we probably don't want a default. If we do, LINK would seems like it would be a more sensible one... Your call here though, we can keep it as null if you think it makes more sense).